sycl/test/basic_tests/SYCL-2020-spec-constants.cpp

```diff
@@ -0,0 +1,98 @@
+// RUN: %clangxx -fsycl -fsycl-device-only -c -o %t.bc %s
+// RUN: sycl-post-link %t.bc -spec-const=rt -o %t-split1.txt
+// RUN: cat %t-split1_0.prop | FileCheck %s
+// RUN: sycl-post-link %t.bc -spec-const=default -o %t-split2.txt
+// RUN: cat %t-split2_0.prop | FileCheck %s
+// RUN: llvm-spirv -o %t-split1_0.spv -spirv-max-version=1.1 -spirv-ext=+all %t-split1_0.bc
+// RUN: llvm-spirv -o %t-split2_0.spv -spirv-max-version=1.1 -spirv-ext=+all %t-split2_0.bc
+//
+//==----------- spec_const.cpp ---------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The test checks that the tool chain correctly identifies all specialization
+// constants, emits correct specialization constats map file and can properly
+// translate the resulting bitcode to SPIR-V.
+
+#include <CL/sycl.hpp>
+
+#include <cstdint>
+#include <vector>
+
+// FIXME: there is a bug with generation SPIR-V friendly IR for boolean spec
+// constants, which was likely interoduced by https://github.com/intel/llvm/pull/3513
+//constexpr sycl::specialization_id<bool> bool_id;
+constexpr sycl::specialization_id<int8_t> int8_id(1);
+constexpr sycl::specialization_id<uint8_t> uint8_id(2);
+constexpr sycl::specialization_id<int16_t> int16_id(3);
+constexpr sycl::specialization_id<uint16_t> uint16_id(4);
+constexpr sycl::specialization_id<int32_t> int32_id(5);
+constexpr sycl::specialization_id<uint32_t> uint32_id(6);
+constexpr sycl::specialization_id<int64_t> int64_id(7);
+constexpr sycl::specialization_id<uint64_t> uint64_id(8);
+// FIXME: enable once we support constexpr constructor for half
+// constexpr sycl::specialization_id<half> half_id(9.0);
+constexpr sycl::specialization_id<float> float_id(10.0);
+constexpr sycl::specialization_id<double> double_id(11.0);
+
+struct composite {
+  int a;
+  int b;
+  constexpr composite(int a, int b): a(a), b(b) {}
+};
+
+constexpr sycl::specialization_id<composite> composite_id(12, 13);
+
+class SpecializedKernel;
+
+int main() {
+  sycl::queue queue;
+
+  std::vector<float> vec(1);
+  {
+    sycl::buffer<float, 1> buf(vec.data(), vec.size());
+    queue.submit([&](sycl::handler &h) {
+      auto acc = buf.get_access<sycl::access::mode::write>(h);
+
+      h.single_task<SpecializedKernel>([=](sycl::kernel_handler kh) {
+ //       auto i1 = kh.get_specialization_constant<bool_id>();
```

Comment:
it is necessary ether to add TODO comment here or remove commented line.

Reply:
Fixed in e7c93de50be86d221cd0f04caeaf3f56780cb877